app/uk/gov/hmrc/gform/eval/smartstring/SmartStringEvaluatorFactory.scala

```diff
@@ -82,9 +85,12 @@ private class Executor(
     new MessageFormat(s.rawValue(formModelVisibilityOptics.booleanExprResolver.resolve(_))(l))
       .format(
         s.interpolations(formModelVisibilityOptics.booleanExprResolver.resolve(_))
-          .map {
-            case concatExpr: Concat => formatConcatExpr(concatExpr, markDown)
-            case interpolation      => formatExpr(interpolation, markDown)
+          .map { expr =>
+            val substitutions = ConcatFormatSubstitutions(concat => formatConcatExpr(concat, markDown))
```

Comment:
just noticed that we may move `substitutions` to upper level, no need to create it each time for each expr